File: 06_strings/06_12_sum_of_beauty_of_all_substrings/sum_of_beauty_of_all_substrings.py
def beauty_sum(s: str):
    result = 0
    for i in range(len(s)):
        frequency = [0] * 26
        for j in range(i, len(s)):
            frequency[ord(s[j]) - ord('a')] += 1
            result += max(frequency) - min(x for x in frequency if x)
    return result

def test(input_str):
    result = beauty_sum(input_str)
    print(f"Input: \"{input_str}\" -> Output: {result}")

# Test cases
test("aabcb")      # Output: 5
test("aabcbaa")    # Output: 17
test("abcde")      # Output: 0
test("abcabc")     # Output: 5
test("aaaa")       # Output: 0
test("abcdabcd")   # Output: 9

File: 06_strings/06_12_sum_of_beauty_of_all_substrings/sum_of_beauty_of_all_substrings.js
function beautySum(s) {
    let result = 0;
    for (let i = 0; i < s.length; i++) {
        const frequency = Array(26).fill(0);
        for (let j = i; j < s.length; j++) {
            frequency[s.charCodeAt(j) - 'a'.charCodeAt(0)] += 1;
            result += Math.max(...frequency) - Math.min(...frequency.filter(x => x > 0));
        }
    }
    return result;
}

function test(inputStr) {
    const result = beautySum(inputStr);
    console.log(`Input: "${inputStr}" -> Output: ${result}`);
}

// Test cases
test("aabcb");      // Output: 5
test("aabcbaa");    // Output: 17
test("abcde");      // Output: 0
test("abcabc");     // Output: 5
test("aaaa");       // Output: 0
test("abcdabcd");   // Output: 9
